### ISSUE TITLE
Builder: does not accept Fully trusted keys anymore

### ISSUE THREAD

**Issue by deeplow** [QubesOS/qubes-issues#6876]
### Brief summary

The builder fails to clone a signed commits repo.

### Steps to reproduce
Try to build @marmarta's Qubes menu implementation. (note: skip to the "solutions you've tried" -- it's an upstream bug with git I think)

1. Follow the instructions for the [qubes builder](https://www.qubes-os.org/doc/qubes-builder/) with a `builder.config` like:

   ```bash
   GIT_BASEURL ?= https://github.com
   GIT_PREFIX ?= QubesOS/qubes-
   NO_SIGN ?= 1
   USE_QUBES_REPO_VERSION = $(RELEASE)

   GIT_URL_desktop_linux_menu = git@github.com:marmarta/qubes-desktop-linux-menu
   BRANCH_desktop_linux_menu = linter
   LESS_SECURE_SIGNED_COMMITS_SUFFICIENT = desktop-linux-menu

   BACKEND_VMM=xen

   DIST_DOM0 ?= fc32
   DISTS_VM ?=

   COMPONENTS ?= \
       core-admin \
       core-admin-client \
       core-admin-linux \
       desktop-linux-menu \
       builder \
       builder-debian \
       builder-rpm

   # ...
   ```
2. Import Marta's developer key to the builders keychain following [these instructions](https://www.qubes-os.org/doc/qubes-builder/#code-verification-keys-management)
3. Set Marta's key to fully trusted (instead of ultimately trusted). According to [the docs](https://www.qubes-os.org/doc/qubes-builder/#code-verification-keys-management) it should be possible:
   ```bash
   # here use "trust" command to set key fully or ultimately trusted - only those keys are accepted by QubesBuilder
   ```
4.  `make get-sources`

### Expected behavior

```
[...]

-> Updating sources for desktop-linux-menu...
--> Fetching from git@github.com:marmarta/qubes-desktop-linux-menu linter...
--> Verifying tags or commits...
---> No tag pointing at c0b47d45562abe9fa6f2b8c93f3e62a9ffa4cb6e
---> c0b47d45562abe9fa6f2b8c93f3e62a9ffa4cb6e does not have a signed tag
---> However, it is signed by a trusted key, and $CHECK is set to signed-tag-or-commit
---> Accepting it anyway
--> Merging...
```

### Actual behavior
```
[...]

-> Updating sources for desktop-linux-menu...
--> Fetching from git@github.com:marmarta/qubes-desktop-linux-menu linter...
--> Verifying tags or commits...
---> No tag pointing at c0b47d45562abe9fa6f2b8c93f3e62a9ffa4cb6e
make: *** [Makefile:218: desktop-linux-menu.get-sources] Error 1
```

### Solutions you've tried

Setting trust to fully -- which worked.

**Comment by deeplow** [comment]
To understand why this happens, see: https://gist.github.com/deeplow/e70f4e617ce3c0fd59b947f0dce60c1f <- this gist demonstrates the issue by running similar commands to the qubes-builder [in the relevant section](https://github.com/QubesOS/qubes-builder/blob/824404c483287f89a021a453f9163f0ce3744b75/scripts/verify-git-tag#L92). 

**TL;DR** there is probably a bug in git that makes it show the key as "UNDEFINED" in trust when it's anything but fully trusted. (or maybe it's by design).

Instead of addressing this as an upstream bug, my suggestion is to simply update the docs and code so that only fully trusted are accepted (I can make PRs for both). Relevant parts to be changed
  - code: https://github.com/QubesOS/qubes-builder/blob/824404c483287f89a021a453f9163f0ce3744b75/scripts/verify-git-tag#L92
  - documentation: https://github.com/QubesOS/qubes-doc/blame/d0bc9e70a22450425bc0b32d238c64c389163baf/developer/building/qubes-builder.md#L202

**Comment by deeplow** [comment]
@andrewdavidwong I have diagnosed this, I think.
